etcdserver/api/v3rpc/util.go

```diff
@@ -81,3 +81,12 @@ func togRPCError(err error) error {
 	}
 	return grpcErr
 }
+
+func isClientCtxErr(ctxErr error, err error) bool {
+	if ctxErr != nil {
+		return ctxErr == context.Canceled || ctxErr == context.DeadlineExceeded
+	}
+	ev, _ := status.FromError(err)
```

Comment:
when there is an error, ev will be nil. if we try to get ev.Code on nil, it might panic. see the code above to improve that.

Reply:
@xiang90  thanks!! Sure, agree. As you mentioned and just looking at status doc, since FromError can return nil we should handle it. I didn't think of it seeing few other places we aren't checking for nil on status.FromError (e.g in client.go and other places) and now it seems like we should, but that's something I will do in separate PR and see if sense that makes sense. Thanks!

Reply:
(I guess we can check on nil or boolean.)